contracts/assets/CommunityERC20.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Mozilla Public License 2.0
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+contract CommunityERC20 is
+    Context,
+    Ownable,
+    ERC20
+{
+    /**
+     * If we want unlimited total supply we should set maxSupply to 2^256-1.
+     */
+    uint256 public maxSupply;
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint256 _maxSupply
+    ) ERC20(_name, _symbol) {
+        maxSupply = _maxSupply;
+    }
+
+    // Events
+
+    // External functions
+
+    function setMaxSupply(uint256 newMaxSupply) external onlyOwner {
+        require(newMaxSupply >= totalSupply(), "MAX_SUPPLY_LOWER_THAN_TOTAL_SUPPLY");
+        maxSupply = newMaxSupply;
+    }
```

Comment:
Are we sure we want `maxSupply` to be changable after deployment?

Reply:
Yes, This is our way of burning functionality. The same idea is implemented in ERC721.
We do not have any free tokens in a bank. We just decrease maxSupply, so less tokens can be distributed.
FYI, I did not implement the smart contract. Andrea did.

Reply:
Ah right, the burn. Gotcha, thanks for reminding